keras_cv/models/mobilenet_v3.py

```diff
@@ -0,0 +1,534 @@
+
+    return apply
+
+
+def InvertedResBlock(
+    expansion, filters, kernel_size, stride, se_ratio, activation, block_id, name=None
+):
+    """An Inverted Residual Block.
+
+    Args:
+        expansion: integer, the expansion ratio, multiplied with infilters to get the
+            minimum value passed to Depth.
+        filters: integer, number of filters for convolution layer.
+        kernel_size: integer, the kernel size for DpethWise Convolutions.
+        stride: integer, the stride length for DpethWise Convolutions.
+        se_ratio: float, ratio for bottleneck filters. Number of bottleneck
+            filters = filters * se_ratio.
+        activation: the activation layer to use.
+        block_id: integer, a unique identification if you want to use expanded
+            convolutions.
+        name: string, layer label.
+
+    Returns:
+        a function that takes an input Tensor representing a InvertedResBlock.
+    """
+    if name is None:
+        name = f"inverted_res_block_{backend.get_uid('inverted_res_block')}"
+
+    def apply(x):
+        shortcut = x
+        prefix = "expanded_conv/"
+        infilters = backend.int_shape(x)[channel_axis]
+
+        if block_id:
+            prefix = "expanded_conv_{}/".format(block_id)
+
+            x = layers.Conv2D(
+                Depth()(infilters * expansion),
+                kernel_size=1,
+                padding="same",
+                use_bias=False,
+                name=prefix + "expand",
+            )(x)
+            x = layers.BatchNormalization(
+                axis=channel_axis,
+                epsilon=1e-3,
+                momentum=0.999,
+                name=prefix + "expand/BatchNorm",
+            )(x)
+            x = activation(x)
+
+        x = layers.DepthwiseConv2D(
+            kernel_size,
+            strides=stride,
+            padding="same" if stride == 1 else "valid",
+            use_bias=False,
+            name=prefix + "depthwise",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name=prefix + "depthwise/BatchNorm",
+        )(x)
+        x = activation(x)
+
+        if se_ratio:
+            with custom_object_scope({"hard_sigmoid": HardSigmoid()}):
+                x = SqueezeAndExcite2D(
+                    filters=Depth()(infilters * expansion),
+                    ratio=se_ratio,
+                    squeeze_activation="relu",
+                    excite_activation="hard_sigmoid",
+                )(x)
+
+        x = layers.Conv2D(
+            filters,
+            kernel_size=1,
+            padding="same",
+            use_bias=False,
+            name=prefix + "project",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name=prefix + "project/BatchNorm",
+        )(x)
+
+        if stride == 1 and infilters == filters:
```

Comment:
is there ever a case where infilters != filters that is a valid case?

Reply:
if not, lets raise an error in that case

Reply:
Yes, infilters != filters would still be a valid case